server/src/endpoints/common.rs

```diff
@@ -191,16 +191,21 @@ where
 
     let cloned_meta = Arc::new(meta.clone());
     let event_id = Rc::new(Mutex::new(None));
+    let is_event = Rc::new(Mutex::new(true));
 
     let future = project_manager
         .send(GetProject { id: project_id })
         .map_err(BadStoreRequest::ScheduleFailed)
-        .and_then(clone!(event_id, |project| {
+        .and_then(clone!(event_id, is_event, |project| {
             extract_envelope(&request, meta)
                 .into_future()
                 .and_then(clone!(project, |envelope| {
                     *event_id.lock() = Some(envelope.event_id());
 
+                    if envelope.get_item_by_type(ItemType::Event).is_none() {
+                        *is_event.lock() = false;
```

Comment:
This missed certain event types. Security reports and minidumps also need outcomes, but have no event Iran at this stage. 

Can we inject this flag statically from the endpoint, instead?

Reply:
I think we need the dynamic behavior for store and static behavior for all other endpoints.

Reply:
Agreed, although I’m not happy with the way this is achieved at the moment. I think we might need to spend some more time thinking about how to achieve this properly. Let’s tackle this in a follow-up PR. 

Would you mind backing this out for now?

Reply:
kk it's just based on endpoint now